src/GraphQL/Types/DecimalGraphType.cs

```diff
@@ -4,8 +4,6 @@ namespace GraphQL.Types
 {
     public class DecimalGraphType : ScalarGraphType
     {
-        public DecimalGraphType() => Name = "Decimal";
```

Comment:
seems these changes are unrelated to the long change? perhaps better as another PR?

Reply:
These changes are quite simple and obvious. Even last time when new scalars were added I wanted to do that. The appearance of Long was a good occasion.

Reply:
I'll also note that I'm not a fan of this change myself and may just reverse it.  I would rather it be very explicit what the name is and not use conventions.  Conventions like this is often where people start to get confused and do the wrong thing (not even realizing that they are _required_ provide a name), then try to force the framework to do something it was not intended to do.

Reply:
@joemcbride  First of all, the default constructor’s goal was to give the type name a value other than `null`. Due to the absence of restrictions / checks on this and other properties of the schema, a number of errors could occur. See #1284 #1156 and related issues. The type must have a name. But it’s very easy to forget to set it. Also related to #1257.

> not even realizing that they are _required_ provide a name

Well, even before this change, they did not realize it. They **could** not set a name and subsequently face various kinds of incomprehensible behavior. So the situation has not gotten worse. Don't you think so?

Reply:
> Due to the absence of restrictions / checks on this

So let's fix that real issue instead of making work-arounds for it.

> So the situation has not gotten worse.

It's worse in my book since now they won't be getting an error.  I prefer that they get an error over trying to teach and support a convention.  I've had to work with frameworks that use conventions heavily and it's a huge learning curve for developers and they forget half the time.  I would rather it just be explicit.

Reply:
Let's say that this is a weighty argument and I also partially dislike conventions. But what about [this](https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL/Types/GraphType.cs#L36) then? These are the same conventions. When I figured out why the above errors occurred, I just discovered this place where the type name is calculated. Apparently, the type name [was set](https://github.com/graphql-dotnet/graphql-dotnet/blob/master/src/GraphQL/Types/GraphTypesLookup.cs#L234) later than the code that caused the error. I thought so - since here the name is already assigned (by conventions from GetType().Name) then why not make it at the earliest moment - in ctor?